app/rest/connector-generator/src/main/java/io/syndesis/connector/generator/swagger/SwaggerHelper.java

```diff
@@ -33,4 +67,73 @@ private SwaggerHelper() {
             throw new IllegalStateException("Unable to serialize Swagger specification", e);
         }
     }
+
+    /* default */ static SwaggerModelInfo parse(final String specification, boolean validate) {
+        final SwaggerParser parser = new SwaggerParser();
+        final Swagger swagger = ofNullable(parser.read(specification)).orElseGet(() -> parser.parse(specification));
+        if (swagger == null) {
+            LOG.debug("Unable to read Swagger specification\n{}\n", specification);
+            throw new IllegalArgumentException(
+                "Unable to read Swagger specification from: " + ofNullable(specification).map(s -> StringUtils.abbreviate(s, 100)).orElse(""));
```

Comment:
Now that we have errors/warnings perhaps we could add this as an error, rather than throwing an exception?